core-web/apps/dotcms-ui/src/app/portlets/dot-containers/container-list/container-list.component.ts

```diff
@@ -1,8 +1,225 @@
+import { DotRouterService } from '@services/dot-router/dot-router.service';
+import { ActionHeaderOptions } from '@models/action-header';
+import { DataTableColumn } from '@models/data-table';
+import { DotContainer } from '@models/container/dot-container.model';
+import { DotContentState } from '@dotcms/dotcms-models';
+import {
+    DotContainerListState,
+    DotContainerListStore
+} from '@portlets/dot-containers/container-list/store/dot-container-list.store';
 
 @Component({
     selector: 'dot-container-list',
     templateUrl: './container-list.component.html',
-    styleUrls: ['./container-list.component.scss']
+    styleUrls: ['./container-list.component.scss'],
+    providers: [DotContainerListStore]
 })
-export class ContainerListComponent {}
+export class ContainerListComponent implements OnInit, OnDestroy {
+    @ViewChild('listing', { static: false })
+    vm$ = this.store.vm$;
+
+    listing: DotListingDataTableComponent;
+    selectedContainers: DotContainer[] = [];
+
+    containerBulkActions$: Observable<MenuItem[]>;
+    tableColumns$: Observable<DataTableColumn[]>;
+    actionHeaderOptions$: Observable<ActionHeaderOptions>;
+    addToBundleIdentifier$: Observable<string>;
+
+    updateBundleIdentifier = this.store.updateBundleIdentifier;
+
+    private isEnterPrise: boolean;
+    private hasEnvironments: boolean;
+    private destroy$: Subject<boolean> = new Subject<boolean>();
+
+    constructor(
+        private store: DotContainerListStore,
+        private dotMessageService: DotMessageService,
+        private dotAlertConfirmService: DotAlertConfirmService,
+        private dotPushPublishDialogService: DotPushPublishDialogService,
+        private dotRouterService: DotRouterService
+    ) {}
+
+    ngOnInit(): void {
+        this.store.updateTableColumns(this.setContainerColumns());
+        this.store.updateContainerBulkActions(this.setContainerBulkActions());
+
+        this.containerBulkActions$ = this.vm$.pipe(
+            takeUntil(this.destroy$),
+            pluck('containerBulkActions')
+        );
+
+        this.tableColumns$ = this.vm$.pipe(takeUntil(this.destroy$), pluck('tableColumns'));
+
+        this.actionHeaderOptions$ = this.vm$.pipe(
+            takeUntil(this.destroy$),
+            pluck('actionHeaderOptions')
+        );
+
+        this.addToBundleIdentifier$ = this.vm$.pipe(
+            takeUntil(this.destroy$),
+            pluck('addToBundleIdentifier')
+        );
+
+        this.vm$
+            .pipe(takeUntil(this.destroy$))
+            .subscribe(({ selectedContainers }: DotContainerListState) => {
+                this.selectedContainers = selectedContainers;
+            });
+
+        this.vm$
+            .pipe(takeUntil(this.destroy$))
+            .subscribe(({ isEnterPrise, hasEnvironments }: DotContainerListState) => {
+                this.isEnterPrise = isEnterPrise;
+                this.hasEnvironments = hasEnvironments;
+            });
```

Comment:
Instead of thing all this extra properties you can use the state from the store directly in the template:

```
<ng-container *ngIf="vm$ | async as vm">
    {{vm.containerBulkActions}}
    {{vm. selectedContainers}}
</ng-container>
```

And so on.

https://ngrx.io/guide/component-store/usage

<img width="1189" alt="image" src="https://user-images.githubusercontent.com/751424/190165078-b08ee186-7ff9-4028-8d9a-505e07c9a391.png">